### PR TITLE
fix(docs): comparison

### LINKED ISSUE #1008: Move the docs from Github's wiki to the ./docs directory
It will be way easier for people to search something in the docs if they are part of the repo itself instead of relying on GitHub to host the docs.

### PR DESCRIPTION
**Pull request by dbritto-dev**
## Related Issues

Fixes #1008 

## Summary

- Move difference between zustand and valtio to ./docs
- Updating zustand vs redux
- Adding other comparisons

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

**Review by Exalted100 — COMMENTED** [review]
- For the difference between Zustand and Redux, I think some exist (the differences that were mentioned earlier and removed from the document). While Redux Toolkit removes some differences, it still involves wrapping a context provider around the app and other differences could still apply. The only difference I think Redux toolkit takes away besides that is that it makes Redux almost as simple to implement as Zustand.

Everything else looks fine to me.

**Comment by dbritto-dev** [comment]
@Exalted100 about Redux since we're just comparing Redux (React Redux) with Zustand, anything related to Redux shouldn't be considered as part of this section.

**Comment by Exalted100** [comment]
Since that's the case, we could include the differences from before since they still apply, right?

**Comment by dbritto-dev** [comment]
@Exalted100 yeah with an exception about hooks, because now Redux has support for hooks.

**Comment by dbritto-dev** [comment]
@Exalted100 ready to review :D

File: docs/getting-started/comparison.md
Comment (chrisk-7777) [inline docs/getting-started/comparison.md:189]: nit: I _think_ the ":" isn't required?

File: docs/getting-started/comparison.md
Comment (chrisk-7777) [inline docs/getting-started/comparison.md:100]: minor: As above "difference" -> "differences"

I would maybe reword slightly to make it clear we're talking about optimization. Perhaps something like:

> When it comes to render optimizations within your app, there are no major differences in approach between Zustand and Redux. In both libraries it is recommended you manually render optimizations by using selectors.

File: docs/getting-started/comparison.md
Comment (chrisk-7777) [inline docs/getting-started/comparison.md:222]: I feel like by this point we're repeating this line. 

How would you feel about an opening line, right at the top of the page, with something like:

> Zustand is one of many state management libraries for React. On this page we'll discuss Zustand in comparison to some of these libraries, including Redux, Valtio, Jotai, and Recoil.
>
> Each library has its own strengths and weaknesses, and we will compare key differences and similarities between each. 
>

File: docs/getting-started/comparison.md
Comment (dai-shi) [inline docs/getting-started/comparison.md:71]: This is traditional Redux. If you make a comparison note now, you want to compare with modern Redux.

File: docs/getting-started/comparison.md
Comment (chrisk-7777) [inline docs/getting-started/comparison.md:227]: "consits" -> "consists"

**Review by chrisk-7777 — COMMENTED** [review]
This is looking really good, I just had some minor grammatical suggestions. 
Let me know your thoughts :) 

Also, do you know in Github, will it squash all the WIP commits?

File: docs/getting-started/comparison.md
Comment (dai-shi) [inline docs/getting-started/comparison.md:228]: Another more major difference is Zustand store is global in memory, but Jotai atoms are not. Atoms are definitions that don't hold values. That's why you can't read atom values outside React (as it can be multiple values.)

**Review by dai-shi — COMMENTED** [review]
I'm not sure what kind of comparison we should provide and what readers expect.
I don't have big opinions on what it should be. What I feel is there are many (or mostly) code snippets.
I'd rather let the people who work on docs to discuss.
cc @chrisk-7777

File: docs/getting-started/comparison.md
Comment (dbritto-dev) [reply docs/getting-started/comparison.md:228]: Cool, good to know

File: docs/getting-started/comparison.md
Comment (dbritto-dev) [reply docs/getting-started/comparison.md:71]: What you mean with modern Redux? Actually, Redux toolkit use createStore behind scenes. I can update the code to use the Redux Toolkit version but I want to show the basic example because that's why people often use even these days.

Comment (dai-shi) [reply docs/getting-started/comparison.md:71]: I mean Redux Toolkit. The bare createStore should be marked as deprecated.

Comment (dbritto-dev) [reply docs/getting-started/comparison.md:71]: I added a piece of code for Redux Toolkit as well

**Comment by dbritto-dev** [comment]
Hey folks, ready for review

@dai-shi @chrisk-7777 @Exalted100

File: docs/getting-started/comparison.md
Comment (dai-shi) [inline docs/getting-started/comparison.md:284]: you can with zustand. you can't with jotai.

**Comment by dbritto-dev** [comment]
@dai-shi done

**Review by dai-shi — COMMENTED** [review]
Looks okay to me. I'll wait for others to review.

File: docs/getting-started/comparison.md
Comment (Exalted100) [inline docs/getting-started/comparison.md:280]: It should be "differences", not "difference".

**Review by Exalted100 — COMMENTED** [review]
It looks fine to me. Just one small typo which I left a comment for.

**Comment by dbritto-dev** [comment]
@Exalted100 done

**Review by Exalted100 — APPROVED** [review]

**Comment by dai-shi** [comment]
Let's wait a final call from @chrisk-7777.

File: docs/getting-started/comparison.md
Comment (dai-shi) [inline docs/getting-started/comparison.md:283]:
```suggestion
Jotai atoms are not (are definitions that do not hold values) and that's why
```

**Comment by chrisk-7777** [comment]
Thanks for actioning the feedback @dbritto-dev , I have reread the PR and have no further feedback

**Pull request merged**
